src/bin/cratesfyi.rs

```diff
@@ -297,19 +278,17 @@ enum BuildSubcommand {
 
     /// Unlocks cratesfyi daemon to continue building new crates
     Unlock,
-
-    PrintOptions,
 }
 
 impl BuildSubcommand {
-    pub fn handle_args(self, ctx: BinContext, mut docbuilder: DocBuilder) -> Result<(), Error> {
+    pub fn handle_args(self, ctx: BinContext, skip_if_exists: bool) -> Result<(), Error> {
+        let mut docbuilder = DocBuilder::new(ctx.config()?, ctx.pool()?, ctx.build_queue()?);
+
         let rustwide_builder = || -> Result<RustwideBuilder, Error> {
-            Ok(RustwideBuilder::init(
-                ctx.config()?,
-                ctx.pool()?,
-                ctx.metrics()?,
-                ctx.storage()?,
-            )?)
+            let mut builder =
+                RustwideBuilder::init(ctx.config()?, ctx.pool()?, ctx.metrics()?, ctx.storage()?)?;
```

Comment:
This has the same duplication you removed in the previous commit because it's a closure in a separate function. Could you make `rustwide_builder` a method on `Context` instead so it can be reused here?

Reply:
Oops, you did that in a later commit 😆 `RustwideBuilder::init` taking a context is just as good.